tests/sentry/notifications/platform/api/endpoints/test_internal_registered_templates.py

```diff
@@ -25,3 +27,97 @@ def test_get_registered_templates(self) -> None:
                 serialize_template(template=template, source=source)
                 in response.data[template.category.value]
             )
+
+
+class SerializeSlackPreviewTest(TestCase):
+    def setUp(self) -> None:
+        self.template = MockNotificationTemplate()
+
+    def test_serialize_slack_preview_complete(self) -> None:
+        """Test complete serialization of Slack preview with all components"""
+        result = serialize_slack_preview(self.template)
+
+        # Should return a dict with blocks key
+        assert isinstance(result, dict)
+        assert "blocks" in result
+        assert isinstance(result["blocks"], list)
+
+        blocks = result["blocks"]
+        assert len(blocks) >= 4  # header, body, actions, footer, image
+
+        # Check header block
+        header_block = blocks[0]
+        assert header_block["type"] == "header"
+        assert header_block["text"]["type"] == "plain_text"
+        assert header_block["text"]["text"] == "Mock Notification"
```

Comment:
Would it make sense to break this into a helper method to that issues these assertions? Some of them seem similar/potentially reusable.